types/constants.go

```diff
@@ -13,42 +13,113 @@ import (
-	OakHardforkBlock        BlockHeight
-	OakHardforkFixBlock     BlockHeight
+
+	// OakDecayDenom is the denominator for how much the total timestamp is decayed
+	// each step.
+	OakDecayDenom int64
+	// OakDecayNum is the numerator for how much the total timestamp is decayed each
+	// step.
+	OakDecayNum int64
+	// OakHardforkBlock is the height at which the hardfork to switch to the oak
+	// difficulty adjustment algorithm is triggered.
+	OakHardforkBlock BlockHeight
+	// OakHardforkFixBlock is the height at which the hardfork to switch from the broken
+	// oak difficulty adjustment algorithm to the fixed oak difficulty adjustment
+	// algorithm is triggered.
+	OakHardforkFixBlock BlockHeight
+	// OakHardforkTxnSizeLimit is the maximum size allowed for a transaction, a change
+	// which I believe was implemented simultaneously with the oak hardfork.
 	OakHardforkTxnSizeLimit = uint64(64e3) // 64 KB
-	OakMaxBlockShift        int64
-	OakMaxDrop              *big.Rat
-	OakMaxRise              *big.Rat
+	// OakMaxBlockShift is the maximum number of seconds that the oak algorithm will shift
+	// the difficulty.
+	OakMaxBlockShift int64
+	// OakMaxDrop is the drop is the maximum amount that the difficulty will drop each block.
+	OakMaxDrop *big.Rat
+	// OakMaxRise is the maximum amount that the difficulty will rise each block.
+	OakMaxRise *big.Rat
 
-	RootDepth        = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
-	RootTarget       Target
+	// RootDepth is the cumulative target of all blocks. The root depth is essentially
+	// the maximum possible target, there have been no blocks yet, so there is no
+	// cumulated difficulty yet.
+	RootDepth = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
+	// RootTarget is the target for the genesis block - basically how much work needs
+	// to be done in order to mine the first block. The difficulty adjustment algorithm
+	// takes over from there.
+	RootTarget Target
+	// SiacoinPrecision is the number of base units in a siacoin. The Sia network has a very
+	// large number of base units. We call 10^24 of these a siacoin.
+	//
+	// The base unit for Bitcoin is called a satoshi. We call 10^8 satoshis a bitcoin,
+	// even though the code itself only ever works with satoshis.
 	SiacoinPrecision = NewCurrency(new(big.Int).Exp(big.NewInt(10), big.NewInt(24), nil))
-	SiafundCount     = NewCurrency64(10000)
-	SiafundPortion   = big.NewRat(39, 1000)
-	TargetWindow     BlockHeight
+	// SiafundCount is the total number of Siafunds in existence.
+	SiafundCount = NewCurrency64(10000)
+	// SiafundPortion is the percentage of siacoins that is taxed from FileContracts.
+	SiafundPortion = big.NewRat(39, 1000)
+	// TargetWindow {help}
+	TargetWindow BlockHeight
```

Comment:
TargetWindow is the number of blocks to look backwards when determining how much time has passed vs. how many blocks have been created. It's only used in the old, broken difficulty adjustment algorithm.